bspump/elasticsearch/data_feeder.py

```diff
@@ -0,0 +1,47 @@
+import orjson
+
+"""
+Data feeders to be used in ElasticSearchSink.
+"""
+
+
+async def data_feeder_create_or_index(items):
+	for _id, data in items:
+		yield b'{"create":{}}\n' if _id is None else orjson.dumps(
+			{"index": {"_id": _id}}, option=orjson.OPT_APPEND_NEWLINE
+		)
+		yield data
+
+
+async def data_feeder_create(items):
+	for _id, data in items:
+		yield b'{"create":{}}\n' if _id is None else orjson.dumps(
+			{"create": {"_id": _id}}, option=orjson.OPT_APPEND_NEWLINE
+		)
+		yield data
+
+
+async def data_feeder_index(items):
+	for _id, data in items:
+		yield b'{"index":{}}\n' if _id is None else orjson.dumps(
+			{"index": {"_id": _id}}, option=orjson.OPT_APPEND_NEWLINE
+		)
+		yield data
+
+
+async def data_feeder_update(items):
+	for _id, data in items:
+		yield orjson.dumps(
+			{"update": {"_id": _id}}, option=orjson.OPT_APPEND_NEWLINE
+		)
+		update_data = b'{"doc":' + data[:-1] + b'}\n'
```

Comment:
You cannot construct the JSON this way - it is unsecure and unreliable, use `orjson` serializer.